skyportal/utils/offset.py

```diff
@@ -398,14 +401,71 @@ def get_nearby_offset_stars(
     )
 
 
+class Cache:
+    def __init__(self, cache_dir, max_items):
+        cache_dir = Path(cache_dir)
+        if not cache_dir.is_dir():
+            cache_dir.mkdir()
+
+        self._cache_dir = Path(cache_dir)
+        self._max_items = max_items
+
+    def _hash_fn(self, fn):
+        m = hashlib.md5()
+        m.update(fn.encode('utf-8'))
+        return self._cache_dir / f'{m.hexdigest()}'
+
+    def __getitem__(self, name):
+        # Cache is disabled, return nothing
+        if self._max_items == 0:
+            return None
+
+        cache_file = self._hash_fn(name)
+        if not cache_file.exists():
+            return None
+
+        log(f"cache hit [{name}]")
+        cache_file.touch()  # Make newest in cache
+
+        return cache_file
+
+    def __setitem__(self, name, data):
+        # Cache is disabled, do not add entry
+        if self._max_items == 0:
+            return
+
+        fn = self._hash_fn(name)
+        with open(fn, 'wb') as f:
+            f.write(data)
+
+        log(f"cache save [{name}]")
+
+        self.clean_cache()
+
+    def clean_cache(self):
+        # Remove stale cache files
+        cached_files = [
+            (f.stat().st_mtime, f.absolute()) for f in self._cache_dir.glob('*')
```

Comment:
Should we have a namespace for the filename in case we want to use this cache dir for other file types? The * completion is a bit aggressive.

Reply:
Yes, I can add that---and probably should, so we can reuse this for our thumbnails too.  I'll add it to the cache constructor.

Reply:
Decided to handle this, for now, with a different base dir (the finder cache is in `cache/finder`)